openlibrary/catalog/utils/__init__.py

```diff
@@ -77,17 +75,14 @@ def flip_name(name: str) -> str:
     :param str name: e.g. "Smith, John." or "Smith, J."
     :return: e.g. "John Smith" or "J. Smith"
     """
-
     m = re_end_dot.search(name)
     if m:
         name = name[:-1]
     if name.find(', ') == -1:
         return name
-    m = re_marc_name.match(name)
-    if isinstance(m, Match):
+    if m := re_marc_name.match(name):
         return m.group(2) + ' ' + m.group(1)
-
-    return ""
+    return ''
 
 
 def remove_trailing_number_dot(date):
```

Comment:
What does `number` in the method name refer to? It seems like perhaps some since-removed functionality?

Reply:
Looks like it is (still) used in extracting dates from MARC fields

https://github.com/internetarchive/openlibrary/blob/586466c479c6f59da2cb67367f98c498c61c9b2a/openlibrary/catalog/marc/parse.py#L330-L334

I think the records will often have punctuation after the final date. I think the 'number' refers to the regex. It's another example of where a method, its regex and the location it's used are all spread out making it difficult to follow.

Looks like the whole thing could be replaced by a regex in the one place it is used in /catalog/marc/parse.py

Reply:
Ahhh, remove trailing dot from number, not removing trailing number & dot (as I parsed it). Got it!